your_daily_dose_malware/main.py

```diff
@@ -1,12 +1,73 @@
 import click
 from commands.scraper import scraper
+import requests
+from dotenv import load_dotenv
+import os
+load_dotenv()
+@click.command(
+    help="""
+    Download eitheir hundred recent malwares uploaded within the last 60 min or  all the recent malwares uploaded within the last 48 hours sha256 hash by using api from 'https://mb-api.abuse.ch/api/v1/' 
 
+    -s256 or --by-sha256: get the lastest sha256 hashes from 'https://bazaar.abuse.ch/export/txt/sha256/recent' save them in sha256_names.txt then for each file  download all malwares in zip file
 
-@click.group()
-def scraper():
-    """:) Your Daily Dose Malware :)"""
-
+    -hr or --hundred-recent: get the lastest hundred recent malwares uploaded within the last 60 min
+    """)
+@click.option('-s256','--by-sha256',is_flag=True)
+@click.option('-hr','--hundred-recent',is_flag=True)
+def run_scrapper(by_sha256, hundred_recent):
+    headers = {
+        'API-KEY':os.getenv('API_KEY')
+    }
```

Comment:
Probably you want to split this functionality in two methods.
1) to get the recent hashes, and return a list.
2) a method to download and write all the malware samples from the list.
From your run scrapper, you will run the two methods sequentially.

Reply:
It's better like this cause this command is doing to much than is should

Reply:
I will split like it as you said

Reply:
Just one commans but dont put all your logic on run_scrapper, move this to functions.

You will need to maintain multiple backends each with their own list and download implementations.

Reply:
i dont know if its what you want but look at my latest commit i have separate logic into two different functions

Reply:
Its a start. I still think you are repeating yourself too much, but we might rewrite this when we add more backends, to write functionality that would need to be reused.